src/WebJobs.Script.WebHost/Configuration/ExtensionsSystemOptions.cs

```diff
@@ -0,0 +1,87 @@
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Azure.WebJobs.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
+{
+    public sealed class ExtensionsSystemOptions : IOptionsFormatter
+    {
+        private static readonly JsonSerializerOptions Options = CreateJsonOptions();
+
+        /// <summary>
+        /// Gets the per-extension system configuration settings.
+        /// </summary>
+        public IDictionary<string, ExtensionSystemOptions> Extensions { get; internal set; } = new Dictionary<string, ExtensionSystemOptions>();
+
+        /// <summary>
+        /// Returns true if the specified extension allows anonymous access to its webhook endpoint.
+        /// </summary>
+        /// <param name="extensionName">The name of the extension.</param>
+        /// <returns>True if the extension allows anonymous access; otherwise, false.</returns>
+        public bool WebhookAllowsAnonymous(string extensionName)
+        {
+            if (extensionName == null)
+            {
+                throw new ArgumentNullException(nameof(extensionName));
+            }
+
+            if (extensionName.Length == 0)
+            {
+                throw new ArgumentException("Value cannot be an empty string.", nameof(extensionName));
+            }
+
+            if (Extensions.TryGetValue(extensionName, out var extensionOptions))
+            {
+                return extensionOptions.WebhookAuthorizationLevel == AuthorizationLevel.Anonymous;
+            }
+
+            // default to System auth if not specified
+            return false;
+        }
+
+        public string Format()
+        {
+            return JsonSerializer.Serialize(this, Options);
+        }
+
+        private static JsonSerializerOptions CreateJsonOptions()
+        {
+            return new JsonSerializerOptions
+            {
+                WriteIndented = true,
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+            };
+        }
+
+        /// <summary>
+        /// Represents the system options for an individual extension.
+        /// </summary>
+        public class ExtensionSystemOptions
+        {
+            private AuthorizationLevel _webhookAuthorizationLevel = AuthorizationLevel.System;
+
+            /// <summary>
+            /// Gets or sets the default authorization level for the extension. Only applies to WebHook extensions.
+            /// </summary>
+            [JsonConverter(typeof(JsonStringEnumConverter))]
+            public AuthorizationLevel WebhookAuthorizationLevel
+            {
+                get => _webhookAuthorizationLevel;
+                set
+                {
+                    if (value != AuthorizationLevel.System && value != AuthorizationLevel.Anonymous)
```

Comment:
Any reason to disallow the other levels?

Reply:
Yes. Currently this is only used in the System policy and that's used only for WebHooks. That policy only supports System, and now Anonymous.